qiskit/providers/ibmq/ibmqsingleprovider.py

```diff
@@ -111,7 +113,8 @@ def _authenticate(self, credentials):
             if 'License required' in str(ex):
                 # For the 401 License required exception from the API, be
                 # less verbose with the exceptions.
-                root_exception = None
+                raise UnlicensedError("Couldn't connect to IBMQ server: {0}"
```

Comment:
Instead of raising the new `UnlicensedError` exception here, in this case we want to keep this `IBMQSingleProvider` method only emitting `ConnectionError`s upwards. However, what we can do is make use of the new class for identifying the situation in a cleaner way than before, via an extra `except` block:

```python
        # Check if the URL belongs to auth services of the new API.
        try:
            ...existing code
        except  AuthenticationLicenseError as ex:
            # now we can be sure that if we get here, is because of license problems!
            raise ConnectionError(.....) from None
        except Exception as ex:
            ....
```

the `from None` bit is for preventing too verbose output reaching the user - again, partly pending an overhaul during #106 .